flow/record/base.py

```diff
@@ -756,6 +756,30 @@ def fieldtype(clspath):
     return t
 
 
+@functools.lru_cache(maxsize=4069)
+def merge_record_descriptors(descriptors, replace=False, name=None):
+    """Create a newly merged RecordDescriptor from a list of RecordDescriptors.
+    This function uses a cache to avoid creating the same descriptor multiple times.
+
+    Duplicate fields are ignored in `descriptors` unless `replace=True`.
+
+    Args:
+        descriptors (Tuple[RecordDescriptor]): Tuple of RecordDescriptors we use for extending/replacing.
+        replace (bool): if `True`, it will replace existing field names, eg: last descriptor always wins.
+        name (str): rename the RecordDescriptor name to `name`. Otherwise, use name from first descriptor.
+
+    Returns:
+        RecordDescriptor: Merged RecordDescriptor
+    """
+    field_map = collections.OrderedDict()
+    for desc in descriptors:
+        for (ftype, fname) in desc.get_field_tuples():
+            if not replace and fname in field_map:
+                continue
+            field_map[fname] = ftype
+    return RecordDescriptor(name or descriptors[0].name, zip(field_map.values(), field_map.keys()))
```

Comment:
This will error if `descriptors` is empty, maybe put a guard up for that.

Reply:
Refactored it a bit, it can still be `None` which should not be allowed, but this now raises an Exception in RecordDescriptor's __init__().

When #50 is merged, i will rebase and move `RecordDescriptorError` to `flow/record/exceptions.py`